File: solutions/cpp/armstrong-numbers/armstrong_numbers.cpp
#include "armstrong_numbers.h"

#include <string>
#include <cmath>

namespace armstrong_numbers {

    auto is_armstrong_number(int n) -> bool
    {

        auto digits = std::to_string(n);

        int sum = 0;

        for (auto c : digits)
        {
            sum += std::pow(c - '0', digits.size());
        }

        return sum == n;

    }

}  // namespace armstrong_numbers

File: solutions/csharp/armstrong-numbers/ArmstrongNumbers.cs
public static class ArmstrongNumbers
{
    public static bool IsArmstrongNumber(int n)
    {
        var digits = n.ToString();

        int s = 0;

        foreach (char c in digits)
        {
            s += (int)Math.Pow(c - '0', digits.Length);
        }

        return s == n;

    }
}